build.gradle.kts

```diff
@@ -18,35 +11,8 @@ repositories {
 group = Library.group
 version = Library.version
 
-tasks {
-    wrapper {
-        // Steps for upgrading Gradle:
-        // 1. update `gradleVersion` and `distributionSha256Sum`
-        //    (use 'Complete (-all) ZIP Checksum' found here: https://gradle.org/release-checksums/)
-        // 2. run `./gradlew wrapper`
-        //    (will update 'gradle/wrapper/gradle-wrapper.properties')
-        // 3. run `./gradlew wrapper` again
-        //    (might update 'gradle/wrapper/gradle-wrapper.jar', 'gradlew' and 'gradlew.bat')
-        // 4. commit all changes
-
-        gradleVersion = "7.5.1"
-        distributionType = ALL
-        distributionSha256Sum = "db9c8211ed63f61f60292c69e80d89196f9eb36665e369e7f00ac4cc841c2219"
-    }
-
-    val dokkaOutputDir = rootProject.projectDir.resolve("dokka")
-
-    clean {
-        delete(dokkaOutputDir)
-    }
-
-    dokkaHtmlMultiModule {
-        dependsOn(clean)
-        failOnWarning.set(true)
-        outputDirectory.set(dokkaOutputDir)
-    }
-}
-
-apiValidation {
-    ignoredProjects += listOf("ksp-annotations", "ksp-processors")
+tasks.wrapper {
+    gradleVersion = libs.versions.gradle.get()
+    distributionType = ALL
+    distributionSha256Sum = libs.versions.gradleChecksum.get()
```

Comment:
Idk whether this is even needed, to update Gradle your should just call `./gradlew wrapper --gradle-version=<version>`

I don't think we need the checksum here, calling the task should save that to the `gradle-wrapper.properties` automatically

Reply:
This way you only have to update `gradle`, `gradleChecksum` and run `./gradlew wrapper`.
To achieve the same without you would have to run `./gradlew wrapper --gradle-version <version> --distribution-type all --gradle-distribution-sha256-sum <checksum>`.

But I think you are right, maybe we should only keep:
```kotlin
tasks.wrapper {
    distributionType = ALL
}
```
Then you would only have to run `./gradlew wrapper --gradle-version <version> --gradle-distribution-sha256-sum <checksum>`.

Reply:
It doesn't update the checksum if you update to a new version, that sounds odd?